config/kube.py

```diff
@@ -161,42 +244,50 @@ def _troubleshoot_tips():
     log.log_error("Refer file {} for troubleshooting tips".format(fname))
 
 
-def _do_join(server, insecure):
-    try:
-        _download_file(server, insecure)
+def _copy_cni_files():
+    if not os.path.exists(CNI_NET_DIR):
+        os.makedirs(CNI_NET_DIR)
+
+    lst = os.listdir(SONIC_TEMPLATES_DIR)
+    for l in lst:
+        if l.startswith(KUBE_CNI_FNAME_PREFIX):
+            spath = os.path.join(SONIC_TEMPLATES_DIR, l)
+            dpath = os.path.join(CNI_NET_DIR, l[len(KUBE_CNI_FNAME_PREFIX):])
+            shutil.copyfile(spath, dpath)
 
+
+def _do_join(state_db, server):
+    try:
         clicommon.run_command("systemctl enable kubelet")
 
         clicommon.run_command("modprobe br_netfilter")
 
         clicommon.run_command(KUBEADM_JOIN_CMD.format(
             KUBE_ADMIN_CONF, device_info.get_hostname()), ignore_error=True)
 
-        if _is_connected(server):
+        if is_connected(server):
             labels = _get_labels()
             for label in labels:
                 _label_node(label)
 
```

Comment:
All feature enabled labels need to be recreated upon join

Reply:
This code no more. All labels are recorded in state-db and ctrmgrd daemon invokes kube commands to push. BTW, the labels are re-added upon join.